sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -188,24 +580,116 @@ public virtual bool ValidateToken(IReadOnlyList<AttestationSigner> attestationSi
+            {
+                string bodyString = JsonSerializer.Serialize(body);
+
+                // Either base64 or json encode the policy depending on the setting of the encodePolicyBody parameter.
+                encodedDocument = Base64Url.EncodeString(bodyString);
+            }
+            else
+            {
+                encodedDocument = string.Empty;
+            }
+
+            returnValue += encodedDocument;
+            returnValue += ".";
+
+            return returnValue;
+        }
+
+        /// <summary>
+        /// Create a secured JSON Web Token based on the specified token body and the specified signing key.
+        /// </summary>
+        /// <param name="body">Object to be embeeded as the body of the attestation token.</param>
+        /// <param name="signingKey">Key used to sign the attestation token.</param>
+        /// <returns>Returns a secured JWT whose body is the serialized value of the <paramref name="body"/> parameter.</returns>
+        private static string GenerateSecuredJsonWebToken(object body, TokenSigningKey signingKey)
+        {
+            Argument.AssertNotNull(signingKey, nameof(signingKey));
+
+            AsymmetricAlgorithm signer;
+            if (signingKey.Certificate.HasPrivateKey)
+            {
+                signer = signingKey.Certificate.PrivateKey;
+            }
+            else
+            {
+                signer = signingKey.Signer;
+            }
+
+            JsonWebTokenHeader header = new JsonWebTokenHeader
+            {
+                Algorithm = signingKey.Signer is RSA ? "RSA256" : (signingKey.Signer is ECDsa ? "ECDH256" : null),
+                X509CertificateChain = new string[] { Convert.ToBase64String(signingKey.Certificate.Export(X509ContentType.Cert)) },
+            };
+            var serializationOptions = new JsonSerializerOptions
+            {
+                IgnoreNullValues = true,
+            };
+            byte[] jwtHeader = JsonSerializer.SerializeToUtf8Bytes<JsonWebTokenHeader>(header, serializationOptions);
+            string encodedHeader = Base64Url.Encode(jwtHeader);
+
+            byte[] jwtBody = null;
+            if (body != null)
+            {
+                jwtBody = JsonSerializer.SerializeToUtf8Bytes(body, serializationOptions);
+            }
+
+            string encodedBody = jwtBody != null ? Base64Url.Encode(jwtBody) : string.Empty;
+
+            string signedData = encodedHeader + '.' + encodedBody;
+
+            byte[] signature;
+            if (signingKey.Signer is RSA rsaKey)
+            {
+                signature = rsaKey.SignData(Encoding.UTF8.GetBytes(signedData), HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+            }
+            else if (signingKey.Signer is ECDsa ecdh)
+            {
+                signature = ecdh.SignData(Encoding.UTF8.GetBytes(signedData), HashAlgorithmName.SHA256);
+            }
+            else
+            {
+                throw new JsonException();
```

Comment:
Should we consider a description to help callers understand what may have gone wrong, or do we not know ourselves in this case?